From 0ecb7564ca608b4cb9446166a3e9147e23e3e730 Mon Sep 17 00:00:00 2001
From: Scott Graham <scott.github@h4ck3r.net>
Date: Mon, 30 Jun 2025 12:55:16 -0700
Subject: [PATCH 12/13] rename min() to min_() for Windows, and avoid
 sign/unsigned warning

and (void) some assert-only var uses for warnings too.
---
 copy.c | 10 +++++-----
 gvn.c  |  1 +
 rega.c |  1 +
 3 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/copy.c b/copy.c
index 0e3a104..1a61920 100644
--- a/copy.c
+++ b/copy.c
@@ -110,7 +110,7 @@ usewidthle(Fn *fn, Ref r, int w)
 }
 
 static int
-min(int v1, int v2)
+min_(int v1, int v2)
 {
 	return v1 < v2 ? v1 : v2;
 }
@@ -164,9 +164,9 @@ defwidthle(Fn *fn, Ref r, int w)
 				return 1;
 			if (w < 32) {
 				if (i->op == Osar)
-					w = min(31, w+v);
+					w = min_(31, w+v);
 				else
-					w = min(32, w+v);
+					w = min_(32, w+v);
 			}
 		}
 		return defwidthle(fn, i->arg[0], w);
@@ -188,7 +188,7 @@ defwidthle(Fn *fn, Ref r, int w)
 	if (ext(i, &e)) {
 		if (e.zext && e.usew <= w)
 			return 1;
-		w = min(w, e.nopw);
+		w = min_(w, e.nopw);
 		return defwidthle(fn, i->arg[0], w);
 	}
 
@@ -326,7 +326,7 @@ copyref(Fn *fn, Blk *b, Ins *i)
 
 		/* do not break typing by returning
 		 * a narrower temp */
-		if (KWIDE(i->cls) > KWIDE(t->cls))
+		if ((int)KWIDE(i->cls) > KWIDE(t->cls))
 			return NULL_R;
 
 		w  = Wsb + (i->op - Oextsb);
diff --git a/gvn.c b/gvn.c
index e82b321..26333d6 100644
--- a/gvn.c
+++ b/gvn.c
@@ -219,6 +219,7 @@ assoccon(Fn *fn, Blk *b, Ins *i1)
 	|| (i1->cls == Kw && (int32_t)c.bits.i < 0)) {
 		fail = negcon(i1->cls, &c);
 		assert(fail == 0);
+		(void)fail;
 		op = Osub;
 	}
 
diff --git a/rega.c b/rega.c
index 48d6637..279e207 100644
--- a/rega.c
+++ b/rega.c
@@ -257,6 +257,7 @@ move(int r, Ref to, RMap *m)
 	if (bshas(m->b, r)) {
 		/* r is used and not by to */
 		assert(r1 != r);
+		(void)r1;
 		for (n=0; m->r[n] != r; n++)
 			assert(n+1 < m->n);
 		t = m->t[n];
-- 
2.49.0.vfs.0.4

